modules/autoscaling/main.tf

```diff
@@ -29,21 +31,23 @@ resource "aws_appautoscaling_policy" "table_read_policy" {
 }
 
 resource "aws_appautoscaling_target" "table_write" {
-  count = var.create_autoscaling_target && length(var.autoscaling_write) > 0 ? 1 : 0
+  count = length(var.autoscaling_write) > 0 ? 1 : 0
 
   max_capacity       = var.autoscaling_write["max_capacity"]
   min_capacity       = var.autoscaling_write["min_capacity"]
   resource_id        = "table/${var.table_name}"
   scalable_dimension = "dynamodb:table:WriteCapacityUnits"
   service_namespace  = "dynamodb"
+
+  provider = aws.no-tag
```

Comment:
As discussed in-person, add comment saying that this is a workaround for https://github.com/hashicorp/terraform-provider-aws/issues/31839